src/main/java/org/opensearch/flowframework/workflow/CreateIndexStep.java

```diff
@@ -67,89 +64,66 @@ public PlainActionFuture<WorkflowData> execute(
                             createIndexFuture.onFailure(new FlowFrameworkException(errorMessage, ExceptionsHelper.status(exception)));
                         })
                     );
-                } catch (Exception e) {
+                } catch (IOException ex) {
                     String errorMessage = "Failed to parse and update new created resource";
-                    logger.error(errorMessage, e);
-                    createIndexFuture.onFailure(new FlowFrameworkException(errorMessage, ExceptionsHelper.status(e)));
+                    logger.error(errorMessage, ex);
+                    createIndexFuture.onFailure(new FlowFrameworkException(errorMessage, ExceptionsHelper.status(ex)));
                 }
-            }
-
-            @Override
-            public void onFailure(Exception e) {
-                String errorMessage = "Failed to create an index";
+            }, e -> {
+                String errorMessage = "Failed to create the index " + indexName;
                 logger.error(errorMessage, e);
                 createIndexFuture.onFailure(new FlowFrameworkException(errorMessage, ExceptionsHelper.status(e)));
-            }
-        };
-
-        String index = null;
-        String defaultMappingOption = null;
-        Settings settings = null;
-
-        // TODO: Recreating the list to get this compiling
-        // Need to refactor the below iteration to pull directly from the maps
-        List<WorkflowData> data = new ArrayList<>();
-        data.add(currentNodeInputs);
-        data.addAll(outputs.values());
-
-        try {
-            for (WorkflowData workflowData : data) {
-                Map<String, Object> content = workflowData.getContent();
-                index = (String) content.get(getResourceByWorkflowStep(getName()));
-                defaultMappingOption = (String) content.get(DEFAULT_MAPPING_OPTION);
-                if (index != null && defaultMappingOption != null && settings != null) {
-                    break;
-                }
-            }
-        } catch (Exception e) {
-            String errorMessage = "Failed to find the correct resource for the workflow step " + NAME;
-            logger.error(errorMessage, e);
-            createIndexFuture.onFailure(new FlowFrameworkException(errorMessage, ExceptionsHelper.status(e)));
-        }
-
-        // TODO:
-        // 1. Create settings based on the index settings received from content
-
-        try {
-            CreateIndexRequest request = new CreateIndexRequest(index).mapping(
-                FlowFrameworkIndicesHandler.getIndexMappings("mappings/" + defaultMappingOption + ".json"),
-                JsonXContent.jsonXContent.mediaType()
-            );
-            client.admin().indices().create(request, actionListener);
+            }));
         } catch (Exception e) {
-            logger.error("Failed to find the right mapping for the index", e);
+            createIndexFuture.onFailure(e);
```

Comment:
Do we log this error at some point later? Otherwise we should still log it here then

Reply:
This error message has been moved [here](https://github.com/opensearch-project/flow-framework/pull/574/files#diff-a23faf1a736cd43656c055669b094d0c466334179e7b8e584d41b7f32a9943b9R121). That's how we have it for the workflow steps.

Reply:
missed this with the way the git diff was shown